### PR TITLE
Logic for handling api response. Include products that may come from …

### PR DESCRIPTION
**Pull request by victorjefferies**
Need to check logic, and everything else

### CODE REVIEW

File: components/Results.js [hunk 1]
```diff
@@ -46,26 +47,55 @@ export default class Results extends React.Component {
           </View>
         );
       } else {
-          console.log("There are ..." + this.state.dataSource.product.ingredients_from_palm_oil_tags.length())
+          // console.log(this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_tags )
           //Problem here - the ingredients_from_palm_oil_n field doesn't exist on all records. It seems that ngredients_from_palm_oil_tags works though!
-          if(this.state.dataSource.product.ingredients_from_palm_oil_n == 0){
+          if (this.state.dataSource.product.ingredients_from_palm_oil_n > 0 || (this.state.dataSource.product.ingredients_from_palm_oil_tags != undefined && this.state.dataSource.product.ingredients_from_palm_oil_tags.length > 0)) {
```
Comment (AlexDresco) [inline components/Results.js]: Before merging, could we refactor by creating constants like: 

-   const palm_ingredients_total = this.state.dataSource.product.ingredients_from_palm_oil_n;
-         const palm_ingredients_list = this.state.dataSource.product.ingredients_from_palm_oil_tags;
-         const maybe_total = this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_n;
-         const maybe_list = this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_tags;

I currently have it on my computer like this

File: components/Results.js [hunk 2]
```diff
@@ -46,26 +47,55 @@ export default class Results extends React.Component {
...
             <Text style={styles.text}> {this.state.dataSource.product.product_name} </Text>
             <Text style={styles.text}> There are {this.state.dataSource.product.ingredients_from_palm_oil_n} ingredient from Palm oil in this </Text>
-            <Animatable.Text animation="zoomInUp"><Icon name="thumbs-up" size={150} color="#ffff"/></Animatable.Text>
+            <Animatable.Text animation="zoomInUp"><Icon name="thumbs-down" size={150} color="#ffff"/></Animatable.Text>
           </View>
           );
-          } else {
+          } else if (this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_n > 0 || 
+            
+            ( this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_tabs != undefined
```
Comment (AlexDresco) [inline components/Results.js]: Typo here! tabs instead of tags which makes me think that these lines aren't used?

File: components/Results.js [hunk 3]
```diff
@@ -46,26 +47,55 @@ export default class Results extends React.Component {
           </View>
         );
```
Comment (AlexDresco) [inline components/Results.js]: I found a couple of edge cases: 
If you enter 1 - it should go in this case.

Comment (AlexDresco) [reply components/Results.js]: Check where empty goes - if we enter nothing it should go here but goes to "No information"

File: components/Results.js [hunk 4]
```diff
@@ -14,7 +14,8 @@ export default class Results extends React.Component {
   }
 
   componentDidMount(){
-    barcode = this.props.barcode
+    const barcode = this.props.barcode
+    console.log(barcode)
```
Comment (Din-s1) [inline components/Results.js]: Was this meant to remain on line 18 or am i being a ditz?

**Pull request merged**